Testing/VelaptorTests/Input/MouseTests.cs

```diff
@@ -5,189 +5,169 @@
+            this.mockMouseWheelReactable.VerifyOnce(m => m.Subscribe(It.IsAny<Reactor<(MouseScrollDirection, int)>>()));
         }
 
         [Fact]
-        public void GetState_WhenInvoked_ReturnsCorrectState()
+        public void GetState_WhenGettingMousePosition_ReturnsCorrectResult()
         {
             // Arrange
-            var mouse = new Mouse();
-            IMouseInput<MouseButton, MouseState>.InputStates[MouseButton.LeftButton] = true;
+            IReactor<(int, int)>? reactor = null;
 
-            // Act
-            var actual = mouse.GetState();
+            this.mockMousePosReactable.Setup(m => m.Subscribe(It.IsAny<IReactor<(int x, int y)>>()))
+                .Callback<IReactor<(int x, int y)>>(reactorObj =>
+                {
+                    reactor = reactorObj;
+                });
 
-            // Assert
-            Assert.True(actual.IsLeftButtonDown());
-        }
-
-        [Theory]
-        [InlineData(MouseButton.LeftButton)]
-        [InlineData(MouseButton.RightButton)]
-        [InlineData(MouseButton.MiddleButton)]
-        public void SetState_WithButton_CorrectlySetsState(MouseButton button)
-        {
-            // Arrange
-            var mouse = new Mouse();
+            var mouse = CreateMouse();
 
             // Act
-            mouse.SetState(button, true);
+            reactor?.OnNext((11, 22));
+            var actual = mouse.GetState();
 
             // Assert
-            Assert.True(IMouseInput<MouseButton, MouseState>.InputStates[button]);
+            Assert.Equal(11, actual.GetPosition().X);
+            Assert.Equal(22, actual.GetPosition().Y);
         }
 
-        [Fact]
-        public void SetXPos_WhenInvoked_SetsXPosition()
+        [Theory]
+        [InlineData(0)] // Left
+        [InlineData(1)] // Right
+        [InlineData(2)] // Middle
+        public void GetState_WhenGettingMouseButton_ReturnsCorrectResult(int mouseButtonValue)
         {
             // Arrange
-            var mouse = new Mouse();
-
-            // Act
-            mouse.SetXPos(11);
-            var actual = mouse.GetState().GetX();
+            var mouseButton = (MouseButton)mouseButtonValue;
+            IReactor<(MouseButton, bool)>? reactor = null;
 
-            // Assert
-            Assert.Equal(11, actual);
-        }
+            this.mockMouseBtnReactable.Setup(m => m.Subscribe(It.IsAny<IReactor<(MouseButton, bool)>>()))
+                .Callback<IReactor<(MouseButton, bool)>>(reactorObj =>
+                {
+                    reactor = reactorObj;
+                });
 
-        [Fact]
-        public void SetYPos_WhenInvoked_SetsYPosition()
-        {
-            // Arrange
-            var mouse = new Mouse();
+            var mouse = CreateMouse();
 
             // Act
-            mouse.SetYPos(22);
-            var actual = mouse.GetState().GetY();
+            reactor?.OnNext((mouseButton, true));
```

Comment:
Move this line of code to the arrange section

```suggestion
            reactor?.OnNext((mouseButton, true));
```